### ISSUE TITLE
Duplication check is too permissive and duplicates when it shouldn't.

### ISSUE THREAD

**Issue by NightJar** [silverstripe/silverstripe-versioned#127]
https://github.com/silverstripe/silverstripe-versioned/blob/eba04d8b243afda7e060c7b91f9cee005e6b5c79/src/RecursivePublishable.php#L321

This line checks that we both want to duplicate relations, and that we haven't already set what to duplicate.
In legacy; if we _do_ want to duplicate a _specific type_ of relation, then ideally `$relations` will be an array with field names/keys.
But if e.g. we want to copy _only_ the `many_many` fields, and this particular class does not have any many many relationships defined, then this will be an [empty array](https://github.com/silverstripe/silverstripe-framework/blob/4.1/src/ORM/DataObject.php#L425).

An empty array is 'falseish', but is neither `false`, nor `trueish` as is checked for here.

**TL:DR;**
So we ask to copy only `many_many` fields, but end up having _every single relationship_ duplicated instead.

This is rather not ideal for backwards compatibility in that a class that specifies this recently deprecated functionality will also likely [manage it's own duplication methods for such relationships](https://github.com/silverstripe/silverstripe-userforms/blob/5.0.0/code/Model/EditableFormField/EditableMultipleOptionField.php#L123).
